### DISCUSSION TITLE
wpantools for thread network

### DISCUSSION THREAD

**Discussion by sohaibbinaltaf** [openthread/openthread#8783] [Q&A] [answered]
I have built a thread network using raspberry pi 4B and Nordic nRF52840 dongle, with MQTT. Can I use wpan-tools in this setup to get network performance information? Is there any tutorial available which I can follow to install wpan-tools over my thread device?
 If not, 
What other options do I have for packet analysis over the mentioned setup? and can you please confirm is wpantund related to this topic or not?

**Comment by jwhui** [comment, marked as answer]
> I have built a thread network using raspberry pi 4B and Nordic nRF52840 dongle, with MQTT. Can I use wpan-tools in this setup to get network performance information? Is there any tutorial available which I can follow to install wpan-tools over my thread device?
If not,

This [wpan-tools](https://linux-wpan.org/wpan-tools.html) is designed to work with the Linux ieee802154 subsystem. OpenThread currently does not support the Linux ieee802154 subsystem.

> What other options do I have for packet analysis over the mentioned setup?

Depending on the kind of packet analysis you are looking for, you can use standard IP networking tools like `iperf`, `tcpdump`, and `wireshark`.

> and can you please confirm is wpantund related to this topic or not?

This is not related to `wpantund`, which is network interface driver for using OpenThread in NCP mode.

**Reply by sohaibbinaltaf** [reply]
Thanks for clarifying.

**Comment by sohaibbinaltaf** [comment]
I used iperf3 for network performance analysis of my thread network based on raspberry pi 4B and Nordic nRF52840 dongle. I noticed that iperf and ping would only work when one of the thread devices in the network is connected to the ethernet. If I take out the ethernet cable the pinging and iperf stops. 

I would like to know if this is how iperf work, i.e ethernet connectivity is a must or there is something wrong in my configuration? 

secondly, if iperf must require ethernet to be plugged to any one of the thread nodes, does it have any influence on the performance of thread. 

I am using the ipv6 addresses of the thread devices and I am varying -b and -l options to evaluate the performance in different scenarios.

**Reply by jwhui** [reply]
If you are running `iperf` on the RPi, there should not be any requirement to have Ethernet plugged in.

**Reply by sohaibbinaltaf** [reply]
Yes, whole setup is based on RPi's. Is there any way to investigate why this is happening and how to resolve this?

**Reply by jwhui** [reply]
Try pinging using different IPv6 addresses with different scopes (e.g. link-local, mesh-local, vs. global). Certainly, link-local and mesh-local scope should have no bearing on whether there are other network interfaces.
